util-images/request-benchmark/main.go

```diff
@@ -39,18 +39,58 @@ const (
 	NamespaceTmpl       = "%namespace%"
 )
 
+type ContentType string
+
+const (
+	JSONContentType  ContentType = "json"
+	ProtoContentType ContentType = "proto"
+	CBORContentType  ContentType = "cbor"
+	YAMLContentType  ContentType = "yaml"
+)
+
 var (
-	inflight  = flag.Int("inflight", 1, "Benchmark inflight (number of parallel requests being made to the apiserver")
-	namespace = flag.String("namespace", "", "Replace %namespace% in URI with provided namespace")
-	URI       = flag.String("uri", "", "Request URI")
-	verb      = flag.String("verb", "GET", "A verb to be used in requests.")
-	qps       = flag.Float64("qps", -1, "The qps limit for all requests")
+	inflight    = flag.Int("inflight", 1, "Benchmark inflight (number of parallel requests being made to the apiserver")
+	namespace   = flag.String("namespace", "", "Replace %namespace% in URI with provided namespace")
+	URI         = flag.String("uri", "", "Request URI")
+	verb        = flag.String("verb", "GET", "A verb to be used in requests.")
+	qps         = flag.Float64("qps", -1, "The qps limit for all requests")
+	contentType = ContentType("json")
 )
 
+func (c ContentType) String() string {
+	return string(c)
+}
+
+func (c *ContentType) Set(value string) error {
+	switch ContentType(value) {
+	case JSONContentType, ProtoContentType, CBORContentType, YAMLContentType:
+		*c = ContentType(value)
+		return nil
+	default:
+		return fmt.Errorf("invalid content type: %s. Must be one of: [json, proto, cbor, yaml]", value)
+	}
+}
+
 func init() {
+	flag.Var(&contentType, "content-type", "Content type for requests (required). Valid values: [json, proto, cbor, yaml]")
 	flag.Parse()
 }
 
+func getContentTypeHeader(ct ContentType) (string, error) {
+	switch ct {
+	case JSONContentType:
+		return "application/json", nil
+	case ProtoContentType:
+		return "application/vnd.kubernetes", nil
```

Comment:
Shouldn't this be `application/vnd.kubernetes.protobuf`?

Also, you might want to use constants from [here](https://github.com/kubernetes/kubernetes/blob/a18b4a8d97151a25c4d8088839d77e6ba482cac3/staging/src/k8s.io/apimachinery/pkg/runtime/types.go#L46-L49) instead to prevent typos like that in the future.

Reply:
Done.